src/Swarm/TUI/View/Objective.hs

```diff
@@ -84,11 +85,11 @@ drawGoalListItem _isSelected e = case e of
   Header gs -> withAttr boldAttr $ str $ show gs
   Goal gs obj -> getCompletionIcon obj gs <+> titleWidget
    where
-    textSource = obj ^. objectiveTeaser <|> obj ^. objectiveId <|> listToMaybe (obj ^. objectiveGoal)
+    textSource = obj ^. objectiveTeaser <|> obj ^. objectiveId <|> listToMaybe (Markdown.toText <$> obj ^. objectiveGoal)
     titleWidget = maybe (txt "?") withEllipsis textSource
 
 singleGoalDetails :: GoalEntry -> Widget Name
 singleGoalDetails = \case
-  Goal _gs obj -> displayParagraphs $ obj ^. objectiveGoal
+  Goal _gs obj -> displayParagraphs . map Markdown.toText $ obj ^. objectiveGoal
```

Comment:
This needs to change to properly support the whole markdown text (with code) instead of a list of paragraphs.

I suppose it might make sense to use `[[StreamNode]]` as paragraphs so that we can add padding.